_pr-doc-deployment/action.yml

```diff
@@ -199,36 +205,72 @@ runs:
-      id: create-pr-directory
+    - name: "Ensure a clean sub-directory corresponding to the current PR and count number of current deployments"
+      id: create-pull-directory
       shell: bash
       env:
         PR_NUMBER: ${{ github.event.pull_request.number }}
       run: |
         if [[ -d pull/${PR_NUMBER} ]]; then
+          # This means this is a redeployment
+          echo "FIRST_TIME_DEPLOYMENT=false" >> ${GITHUB_OUTPUT}
+
           echo "Directory corresponding to this PR's documentation detected, cleaning ..."
-          rm -rf pull/${PR_NUMBER} && mkdir pull/${PR_NUMBER}
-        else
+          rm -rf pull/${PR_NUMBER}
+
+          # Count the number of deployments before creating the sub-directory
+          # This ensures correct behaviour in the case of redeployments
+          echo "DEPLOYMENT_COUNT=$(ls pull/ | wc -l)" >> ${GITHUB_OUTPUT}
+
           mkdir pull/${PR_NUMBER}
-          echo "Directory for deploying documentation of the PR created."
+        else
           # This means this is the first time the PR documentation is being deployed
           echo "FIRST_TIME_DEPLOYMENT=true" >> ${GITHUB_OUTPUT}
+
+          mkdir pull/${PR_NUMBER}
+          echo "Directory for deploying documentation of the PR created."
+
+          # Count the number of deployments after creating the sub-directory
+          # This ensures accurate behaviour for first time deployments
+          echo "DEPLOYMENT_COUNT=$(ls pull/ | wc -l)" >> ${GITHUB_OUTPUT}
         fi
 
+    - name: "Detect if deployment rejection comment had been previously added to the same PR"
+      if: ${{ fromJSON(steps.create-pull-directory.outputs.DEPLOYMENT_COUNT) > fromJSON(inputs.maximum-pr-doc-deployments) }}
+      id: find-comment
+      uses: peter-evans/find-comment@3eae4d37986fb5a8592848f6a574fdf654e61f9e # v3.1.0
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        body-includes: The documentation for this pull request will not be deployed.
+
+    - name: "Make a PR comment stating reason for not deploying pr documentation"
+      if: ${{ fromJSON(steps.create-pull-directory.outputs.DEPLOYMENT_COUNT) > fromJSON(inputs.maximum-pr-doc-deployments) }}
+      uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043 # v4.0.0
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        comment-id: ${{ steps.find-comment.outputs.comment-id }}
+        edit-mode: replace
+        body: >
+          The documentation for this pull request will not be deployed. This is because the maximum number
+          of allowed deployments (${{ inputs.maximum-pr-doc-deployments }}) set for this repository has been
+          reached.
+
     - name: "Download the pr documentation artifact"
+      if: ${{ fromJSON(steps.create-pull-directory.outputs.DEPLOYMENT_COUNT) <= fromJSON(inputs.maximum-pr-doc-deployments) }}
```

Comment:
I wonder if there is another way to stop the run of the actions without having to add a conditional everywhere.

If there was something similar to a "return", that would be great. If not, let's proceed with your implementation.

Reply:
I see... Then, let's leave as it is right now.